app/models/endpoint.rb

```diff
@@ -5,7 +5,7 @@ class Endpoint < ApplicationRecord
 
   has_one :filter, :dependent => :destroy
 
-  accepts_nested_attributes_for :filter, allow_destroy: true
+  accepts_nested_attributes_for :filter
```

Comment:
Why should we restrict existence of endpoints without filters? It doesn't interfere with the process, but can make user's life easier.
If you are worried about the user not understanding that this endpoint will not be hit, we can disable it automatically, or just show the fact in status.

Reply:
Functionally there is no difference between an endpoint with filter with no apps/event_types/levels and an endpoint without a filter. If an endpoint has no filter and the user updates the endpoint with apps/event_types/levels a filter gets automagically created. At this point, the user doesn't need to know that there is some kind of filter object, let alone be allowed to destroy it.

Reply:
filter with no apps/event_types/levels is allow all. Please make sure it's clear to the user.
We don't have a "deny all" notion in filters.

Reply:
Ok, I forgot about this. Rolled that change back